src/components/_global/BalCard/BalCard.vue

```diff
@@ -39,6 +39,13 @@ export default defineComponent({
       validator: (val: string): boolean => {
         return ['', 'none', 'sm', 'md', 'lg', 'xl'].includes(val);
       }
+    },
+    style: {
+      type: Object,
+      default: () => ({})
+    },
+    ref: {
+      type: Object as PropType<Ref<HTMLElement>>
```

Comment:
I don't think these props are needed. If you want to apply a style attribute to a BalCard you can just do:
```html
<BalCard :style="{ color: blue }" />
```
And this will be applied to the root div of the component, so the one you are applying the prop too. You are essentially copying vue behaviour here.